cmd/kube-apiserver/app/server.go

```diff
@@ -346,12 +364,12 @@ func (s *APIServer) Run(_ []string) error {
 	if err != nil {
 		return err
 	}
-	storageVersions := make(map[string]string)
-	if s.StorageVersion == "" {
-		s.StorageVersion = g.Version
+
+	storageVersions := generateStorageVersionMap(s.StorageVersion, s.StorageVersions)
+	if _, found := storageVersions[g.Group]; !found {
+		storageVersions[g.Group] = g.Version
```

Comment:
The defaults should appear as a default value for the command line parameter. That way, you can see what they are. We should not do a software default here, because that makes the final value impossible for a user to figure out (unless they look at source code).